File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Temperature IR Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var tir = new Tinkerforge.BrickletTemperatureIRV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Get current ambient temperature (unit is °C/10)
        tir.getAmbientTemperature(
            function (ambientTemperature) {
                console.log('Ambient Temperature: ' + ambientTemperature/10.0 + ' °C');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );

        // Get current object temperature (unit is °C/10)
        tir.getObjectTemperature(
            function (objectTemperature) {
                console.log('Object Temperature: ' + objectTemperature/10.0 + ' °C');
            },
            function (error) {
                console.log('Error: ' + error);
            }
        );
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, temperature_ir_v2_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Temperature IR Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let tir = TemperatureIrV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current ambient temperature.
    let ambient_temperature = tir.get_ambient_temperature().recv()?;
    println!(
        "Ambient Temperature: {} °C",
        ambient_temperature as f32 / 10.0
    );

    // Get current object temperature.
    let object_temperature = tir.get_object_temperature().recv()?;
    println!(
        "Object Temperature: {} °C",
        object_temperature as f32 / 10.0
    );

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
